test/Api.Test/KeyManagement/Controllers/UsersControllerTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿#nullable enable
+using Bit.Api.KeyManagement.Controllers;
+using Bit.Api.KeyManagement.Queries;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.KeyManagement.Enums;
+using Bit.Core.KeyManagement.Models.Data;
+using Bit.Core.Repositories;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using NSubstitute;
+using NSubstitute.ReturnsExtensions;
+using Xunit;
+
+namespace Bit.Api.Test.KeyManagement.Controllers;
+
+[ControllerCustomize(typeof(UsersController))]
+[SutProviderCustomize]
+[JsonDocumentCustomize]
+public class UsersControllerTests
+{
+    [Theory]
+    [BitAutoData]
+    public async Task GetPublicKey_NotFound_ThrowsNotFoundException(
```

Comment:
Can we add happy path unit test for `GetPublicKey`

Reply:
Added